src/web/components/ApiKeyManagement/KeyItem.tsx

```diff
@@ -1,11 +1,20 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { useState } from 'react';
+
+import { ApiRoleDTO } from '../../../api/entities/ApiRole';
 import { ApiKeyDTO } from '../../../api/services/adminServiceHelpers';
 import { formatUnixDate } from '../../utils/textHelpers';
 import ApiRolesCell from './ApiRolesCell';
+import KeyEditDialog, { OnApiKeyEdit } from './KeyEditDialog';
 
 type KeyItemProps = {
   apiKey: ApiKeyDTO;
+  onEdit: OnApiKeyEdit;
+  availableRoles: ApiRoleDTO[];
 };
-function KeyItem({ apiKey }: KeyItemProps) {
+function KeyItem({ apiKey: apiKeyInitial, onEdit, availableRoles }: KeyItemProps) {
```

Comment:
Unsure what the best practice on this naming is. My thinking here is above this component it knows it just as the api key being passed in but once it's here we want it to be initial since useState is being used.